android/src/main/java/com/swmansion/rnscreens/gamma/tabs/TabsHost.kt

```diff
@@ -370,6 +377,28 @@ class TabsHost(
         }
     }
 
+    private fun updateBadgeAppearance(
+        menuItemIndex: Int,
+        tabScreen: TabScreen,
+    ) {
+        val badgeValue = tabScreen.badgeValue?.toIntOrNull()
+        if (tabScreen.badgeValue != null && badgeValue == null) {
+            Log.e(TAG, "[RNScreens] Android supports only numbers as badge value")
+        }
```

Comment:
I think that sanitization code should be on setter level, or even on view manager level. View should hold only the valid state (of Int type)